src/main/java/org/apache/mesos/scheduler/plan/BlockFactory.java

```diff
@@ -11,6 +12,6 @@
  * detected for a Task.
  */
 public interface BlockFactory {
-    Block getBlock(String taskType, TaskSpecification taskSpecification)
-            throws InvalidRequirementException;
+    Block getBlock(TaskSpecification taskSpecification) throws Block.InvalidException;
+    List<Block> getBlocks(List<TaskSpecification> taskSpecifications) throws Block.InvalidException;
```

Comment:
Any reason for the switch? Looks like DefaultBlockFactory isn't really doing anything to take advantage of this at the moment.